dds/DCPS/security/AuthenticationBuiltInImpl.cpp

```diff
@@ -1023,20 +1023,21 @@ ::CORBA::Boolean AuthenticationBuiltInImpl::return_identity_handle(
   if (local != local_participants_.end()) {
 
     const RemoteParticipantMap::iterator remote = local->second->validated_remotes.find(identity_handle);
-
-    {
-      ACE_Guard<ACE_Thread_Mutex> handshake_data_guard(handshake_mutex_);
-
-      for (HandshakeDataMap::iterator it = handshake_data_.begin(); it != handshake_data_.end(); /* increment in loop*/) {
-        if (it->second.second == remote->second) {
-          handshake_data_.erase(it++);
-        } else {
-          ++it;
+    if (remote != local->second->validated_remotes.end()) {
+      {
```

Comment:
```suggestion
    if (remote != local->second->validated_remotes.end()) {
```
with matching change below...

Reply:
There is one statement outside of the scoped lock.

Reply:
Right, I missed it.